src/hotspot/share/runtime/thread.cpp

```diff
@@ -902,100 +902,23 @@ char java_runtime_version[128] = "";
 char java_runtime_vendor_version[128] = "";
 char java_runtime_vendor_vm_bug_url[128] = "";
 
-// extract the JRE version string from java.lang.VersionProps.java_version
-static const char* get_java_version(InstanceKlass* ik) {
+// Extract version and vendor specific information.
+static const char* get_java_version_info(InstanceKlass* ik,
+                                         Symbol* field_name,
```

Comment:
Instead of Symbol* field name, can you pass const char * and use
TempNewSymbol field_name = StringTable::new_symbol(name);
and then these can be removed from vmSymbols.hpp also?

Reply:
I think it's better to keep these inside vmSymbols. That way they will be stored inside the CDS archive, and you don't need to create them at runtime. This will save a tiny bit of startup time.

Reply:
Ok, sure.